src/__tests__/initRules.test.js

```diff
@@ -2419,6 +2671,96 @@ describe('initRules', function() {
       );
+
+      it('logs an error when a condition module exceeds the timeout', function(done) {
+        var rules = setupRules([
+          {
+            conditions: [
+              generateCondition(
+                'Condition1',
+                function(module) {
+                  module.exports = function() {
+                    return new Promise(function(resolve) {
+                      setTimeout(resolve, 3000);
+                      windowSetTimeout(function() {
+                        jasmine.clock().tick(2000);
+                      }, 0);
+                    });
+                  };
+                },
+                { timeout: 2000 }
+              )
+            ]
+          }
+        ]);
+
+        var lastPromiseInQueue = initRules(
+          _satellite,
+          rules,
+          moduleProvider,
+          replaceTokens,
+          true
+        );
+
+        lastPromiseInQueue.then(function() {
+          var errorMessage = logger.error.calls.mostRecent().args[0];
+          var expectedErrorMessage =
+            'Failed to execute Condition1 for Test Rule 1 rule. A timeout occurred because the ' +
+            'condition took longer than 2 seconds to complete.';
+          expect(errorMessage).toStartWith(expectedErrorMessage);
+          expect(notifyMonitors).toHaveBeenCalledWith('ruleConditionFailed', {
+            rule: rules[0],
+            condition: rules[0].conditions[0]
+          });
+
+          done();
+        });
+      });
+
+      it('logs an error when an action module exceeds the timeout', function(done) {
+        var rules = setupRules([
+          {
+            actions: [
+              generateAction(
+                'Action1',
+                function(module) {
+                  module.exports = function() {
+                    return new Promise(function(resolve) {
+                      setTimeout(resolve, 3000);
+                      windowSetTimeout(function() {
+                        jasmine.clock().tick(2000);
+                      }, 0);
+                    });
+                  };
+                },
+                { timeout: 2000 }
+              )
+            ]
+          }
+        ]);
+
+        var lastPromiseInQueue = initRules(
+          _satellite,
+          rules,
+          moduleProvider,
+          replaceTokens,
+          true
+        );
+
+        lastPromiseInQueue.then(function() {
+          var errorMessage = logger.error.calls.mostRecent().args[0];
+          var expectedErrorMessage =
+            'Failed to execute Action1 for Test Rule 1 rule. A timeout occurred because the ' +
+            'action took longer than 2 seconds to complete.';
+          expect(errorMessage).toStartWith(expectedErrorMessage);
+          expect(notifyMonitors).toHaveBeenCalledWith('ruleActionFailed', {
+            rule: rules[0],
+            action: rules[0].actions[0]
+          });
+
+          done();
+        });
+      });
     });
   });
 });
```

Comment:
I suggest we start breaking down initRules into smaller modules. The size and complexity of both the module and the spec is unwieldy IMO. Let me know if you want to talk through how you might go about this.